src/main/java/uk/gov/hmcts/reform/pbis/servicebus/MessageQueueProcessor.java

```diff
@@ -77,12 +88,15 @@ private void processMessages(IServiceBusClient serviceBusClient) {
             MessageProcessingResult processingResult = processMessage(message);
             updateMessageInSubscription(message, processingResult, serviceBusClient);
             logProcessingResult(processingResult, message);
+            sendTelemetryDataForMessage(processingResult);
```

Comment:
It looks like `MessageQueueProcessor` emits events that trigger different behaviours (update message, log result, send telemetry) that do not belong directly to the processor itself. These concerns are separate and could be handled in event listeners.

Reply:
keep it simple